src/util/Util.ts

```diff
@@ -83,6 +83,32 @@ export const pipeSafe = <T extends Writable>(readable: NodeJS.ReadableStream, de
   return destination;
 };
 
+/**
+ * Makes sure that listeners always receive the error event of a stream,
+ * even if it was thrown before the listener was attached.
+ * @param stream - Stream that can potentially throw an error.
+ *
+ * @returns The wrapped stream.
+ */
+export const wrapStreamError = <T extends NodeJS.EventEmitter>(stream: T): T => {
+  let error: Error | undefined;
+  let timeout: NodeJS.Timeout;
+  stream.on('error', (err): void => {
+    error = err;
+    timeout = setTimeout((): void => {
+      logger.warn(`No listener was attached but error was thrown: ${err.message}`);
+    }, 1000);
+  });
+  stream.on('newListener', (event: string, listener: (err: Error) => void): void => {
+    if (event === 'error' && error) {
+      clearTimeout(timeout);
```

Comment:
Do we also want to clear the timeout if the error is _not_ defined yet?
Because in that case, the fallback error handler is not needed anymore (and may even be unregistered).

Reply:
The timeout should not be cleared (since it won't be set yet at that point). But there is something to be said for unregistering the fallback error listener if an actual error listener gets attached.

Reply:
^ Yes, that's what should happen.

Reply:
Do we need:
- at start, and on removeListener where no error listeners are left
  - attach fallback error listener
  - attach newListener listener
  - detach removeListener
- on newListener error
  - detach fallback error listener
  - detach newListener listener
  - attach removeListener

For memory reasons, the newListener, removeListener, and error callbacks should be written in such a way that the _same_ function instances are reused for every listener. For that purpose, can store data in `this` (the stream) by using Symbols.